src/transport/manager/mod.rs

```diff
@@ -1706,13 +1713,19 @@ impl TransportManager {
                                         };
                                     }
 
-                                    return Some(TransportEvent::DialFailure {
-                                        connection_id,
-                                        address: Multiaddr::empty(),
-                                        error: Error::Unknown,
-                                    })
+                                    let mut grouped_errors = self.opening_errors.remove(&connection_id).unwrap_or_default();
+                                    grouped_errors.extend(errors);
+                                    return Some(TransportEvent::OpenFailure { connection_id, errors: grouped_errors });
+                                }
+                                Ok(None) => {
+                                    tracing::trace!(
+                                        target: LOG_TARGET,
+                                        ?connection_id,
+                                        "open failure, but not the last transport",
+                                    );
+
+                                    self.opening_errors.entry(connection_id).or_default().extend(errors);
```

Comment:
Just thinking out loud: if the connection fails, we report all the errors for all the addresses, but if the connection succeeds, we don't report the errors for other addresses tried and failed. Feels like a little bit of asymmetry to me. Do we really need to report individual errors for all the addresses?

Reply:
That sounds reasonable, we can probably look into extending a DialSuccess with a list of failed to dial addresses. Don't quite remember if libp2p is doing a similar thing, but that sounds familiar. Yep, wanted to catch a bit more details into why we have seen so many failures from our metrics, the `UnknownError` was a bit too cryptic :D